Extensions/DeviceSensors/JsExtension.js

```diff
@@ -0,0 +1,204 @@
+        ),
+        t("Orientation beta is _PARAM0__PARAM1_"),
+        t("Sensors/Orientation"),
+        "JsPlatform/Extensions/orientation_beta24.png",
+        "JsPlatform/Extensions/orientation_beta32.png"
+      )
+      .addParameter("relationalOperator", t("Sign of the test"))
+      .addParameter("expression", t("Value"))
+      .getCodeExtraInformation()
+      .setIncludeFile(
+        "Extensions/DeviceSensors/devicesensortools.js"
+      )
+      .setFunctionName("gdjs.evtTools.devicesensors.orientation.getOrientationBeta");
+
+      extension
+      .addCondition(
+        "OrientationGamma",
+        t("Compare the value of orientation gamma"),
+        t(
+          "Compare the value of orientation gamma."
+        ),
+        t("Orientation gamma is _PARAM0__PARAM1_"),
+        t("Sensors/Orientation"),
+        "JsPlatform/Extensions/orientation_gamma24.png",
+        "JsPlatform/Extensions/orientation_gamma32.png"
+      )
+      .addParameter("relationalOperator", t("Sign of the test"))
+      .addParameter("expression", t("Value"))
+      .getCodeExtraInformation()
+      .setIncludeFile(
+        "Extensions/DeviceSensors/devicesensortools.js"
+      )
+      .setFunctionName("gdjs.evtTools.devicesensors.orientation.getOrientationGamma");
+
+    extension
+      .addAction(
+        "ActivateOrientationListener",
+        t("Activate orientation sensor"),
+        t("Activate the orientation sensor. (remember to turn it off again)"),
+        t("Activate the orientation sensor."),
+        t("Sensors/Orientation"),
+        "JsPlatform/Extensions/orientation_active24.png",
+        "JsPlatform/Extensions/orientation_active32.png"
+      )
+        .getCodeExtraInformation()
+      .setIncludeFile(
+        "Extensions/DeviceSensors/devicesensortools.js"
+      )
+      .setFunctionName("gdjs.evtTools.devicesensors.orientation.activateOrientationSensor");
+
+    extension
+      .addAction(
+        "DeactivateOrientationListener",
+        t("Deactivate orientation sensor"),
+        t("Deactivate the orientation sensor."),
+        t("Deactivate the orientation sensor."),
+        t("Sensors/Orientation"),
+        "JsPlatform/Extensions/orientation_inactive24.png",
+        "JsPlatform/Extensions/orientation_inactive32.png"
+      )
+        .getCodeExtraInformation()
+      .setIncludeFile(
+        "Extensions/DeviceSensors/devicesensortools.js"
+      )
+      .setFunctionName("gdjs.evtTools.devicesensors.orientation.deactivateOrientationSensor");
+
+    extension
+      .addExpression(
+        "OrientationActive",
+        t("Sensor Active"),
+        t("The expression is true if the device orientation sensor is currently active"),
+        t("Sensors/Orientation")
+      )
+      .getCodeExtraInformation()
+      .setIncludeFile(
+        "Extensions/DeviceSensors/devicesensortools.js"
+      )
+      .setFunctionName("gdjs.evtTools.devicesensors.orientation.isActive");
```

Comment:
So, this works but it's a bit abusing the "type system" of JavaScript/GDevelop as we're returning a boolean instead of a number. This can work but also can lead to really curious error (for example, `1 + true = 2`, but converting if somehow `true` is stored in a variable, and then the variable is converted to a string, then the user will get `true` instead of 1).
So not sure if it's ideal? What you could do is have a "isActiveAsNumber" doing the conversion explicitly (something as simple as `return gdjs.evtTools.devicesensors.orientation._isActive ? 1 : 0`).

Would avoid any edge case in the future :)

Reply:
I decided to go "all number". When the events use numbers anyway there is no need for boolean inside the extension.